lib/rubocop/ast/node.rb

```diff
@@ -510,6 +510,22 @@ def guard_clause?
          (block (send #global_const?({:Class :Module}) :new ...) ...)}
       PATTERN
 
+      def_node_matcher :struct_constructor?, <<~PATTERN
+        (block (send #global_const?(:Struct) :new ...) ...)
+      PATTERN
+
+      def_node_matcher :class_definition?, <<~PATTERN
+        {(class ...)
+         (sclass ...)
+         (casgn _ _ (block (send #global_const?({:Struct :Class}) :new ...) ...))}
+      PATTERN
+
+      def_node_matcher :module_definition?, <<~PATTERN
+        {(module ...)
+         (casgn _ _ (block (send #global_const?(:Module) :new ...) ...))
+         (send nil? :include (block (send #global_const?(:Module) :new ...) ...))}
```

Comment:
Maybe best to simply check for `(block ...`, no? It could be `var = Module.new {}`, `prepend Module.new`, ...

Same for `class_definition?`